# Probability Without Experiment - I

In [1]:
def fact(n):
  if n <= 1:
    return 1
  else:
    return n * fact(n-1)

In [2]:
def per(n, r):
  return int(fact(n) / fact(n-r))

In [3]:
def com(n, r):
  return int(per(n, r) / fact(r))

In [4]:
def px(n, r, p):
  return com(n, r) * (p**r) * (1 - p)**(n - r)

Probability Without Experiment
------------

What is the probability that you would get this combination of balls after 4 trials? 

    One trial = taking out a ball, 
    noting its colour, and 
    putting it back in the bag.

Note that the bag contains 2 blue and 3 red balls.

![ss](https://images.upgrad.com/dc1dc2c5-666e-46ad-8420-47c3955e5c90-pasted%20image%200.png)

    i.e., the combination Blue-Red-Red-Red.

You’re not expected to know the right answer at this point. 

This question is given just to get you thinking in the right direction.

    0.4*0.4*0.4*0.4

    0.6*0.6*0.6*0.6

    0.4*0.6*0.6*0.6

    0.4*0.4*0.6*0.6

---------

    0.4 * 0.6 * 0.6 * 0.6

    ✓ Correct
    Feedback:

The probability of getting a `blue ball` in 1 trial is `0.4`

The probability of getting a `red ball`, as you saw earlier, is `0.6`

Now, if you use the multiplication rule of probability to find the probability for the given combination, you’ll get: 

P(blue ball in first trial) * P(red ball in second trial) * P(red ball in third trial) * P(red ball in fourth trial) 

    = 0.4 * 0.6 * 0.6 * 0.6 = 0.0864

In [5]:
2/5, 3/5

(0.4, 0.6)

In [6]:
2/5 * 3/5 * 3/5 * 3/5

0.08640000000000003

Probability Without Experiment
---------

What is the probability that you will get 3 red balls after 4 trials? 

(One trial = taking out a ball, noting its colour, and putting it back in the bag.)

You’re not expected to know the right answer at this point. 

This question is given just to get you thinking in the right direction.

    0.4*0.6*0.6*0.6

    2*(0.4*0.6*0.6*0.6)

    3*(0.4*0.6*0.6*0.6)

    4*(0.4*0.6*0.6*0.6)

----------

    4 * (0.4 * 0.6 * 0.6 * 0.6)

    ✓ Correct
    Feedback:

Recall that there are four combinations in which you could get three red balls. 

All these four combinations have the same probability, i.e. 0.4*0.6*0.6*0.6. 

So, the total probability of getting 3 red balls is 4*(0.4*0.6*0.6*0.6).

In [7]:
(2/5 * 3/5 * 3/5 * 3/5) + (2/5 * 3/5 * 3/5 * 3/5) + (2/5 * 3/5 * 3/5 * 3/5) + (2/5 * 3/5 * 3/5 * 3/5)

0.34560000000000013

In [8]:
4 * (2/5 * 3/5 * 3/5 * 3/5)

0.34560000000000013

Probability Without Experiment
------------

What is the probability that you would get the combination of balls shown below after 4 trials? 

(One trial = taking out a ball, noting its colour, and putting it back in the bag).
 
![ss](https://images.upgrad.com/42330c5c-adc3-4056-aa0a-b421224600bf-image1.png)

Note that we have changed the contents of the bag — now it has 4 red balls and 1 blue ball.

    0.4 * 0.6 * 0.6 * 0.6

    0.2 * 0.2 * 0.2 * 0.2

    0.8 * 0.8 * 0.8 * 0.8

    0.2 * 0.8 * 0.8 * 0.8

--------------

    0.2 * 0.8 * 0.8 * 0.8

    ✓ Correct
    Feedback:

Since there are 4 red balls and 1 blue ball in the bag now, the probability of getting a `blue ball in 1 trial is 0.2`

The probability of getting a `red ball is 0.8` 

Now, if you use the multiplication rule of probability to find the probability for the given combination, you’ll get: 

P(blue ball in first trial)*P(red ball in second trial)*P(red ball in third trial)*P(red ball in fourth trial) 

    = 0.2 * 0.8 * 0.8 * 0.8 = 0.1024

In [9]:
1/5, 4/5

(0.2, 0.8)

In [10]:
1/5 * (4/5)**3

0.10240000000000003

In [11]:
0.2 * 0.8 * 0.8 * 0.8

0.10240000000000003

Probability Without Experiment
----------

What is the probability that you would get exactly 3 red balls after 4 trials?

(One trial = taking out a ball, noting its colour, and putting it back in the bag.)

    0.2 * 0.8 * 0.8 * 0.8

    2 * (0.2 * 0.8 * 0.8 * 0.8)

    3 * (0.2 * 0.8 * 0.8 * 0.8)

    4 * (0.2 * 0.8 * 0.8 * 0.8)

---------

    4 * (0.2 * 0.8 * 0.8 * 0.8)

    ✓ Correct
    Feedback:

Recall that there are four combinations in which you could get three red balls. 

All these four combinations have the same probability, 

    i.e. 0.2 * 0.8 * 0.8 * 0.8

So, the total probability of getting 3 red balls is 

    4 * (0.2 * 0.8 * 0.8 * 0.8)

In [12]:
f'4 * ({1/5} * {4/5} * {4/5} * {4/5})'

'4 * (0.2 * 0.8 * 0.8 * 0.8)'

# Binomial Distribution

P ( X = r ) = n C r * ( p ) ** r * ( 1 − p ) ** (n − r)

Generalised Probability
------------

What is the probability that you would get 4 red balls after 4 trials 

(One trial = taking out a ball, noting its colour, and putting it back in the bag), 

if the probability of getting a red ball in one trial is equal to p?

    4*p

    p**4

    4p**3

    p**3 * (1−p)

----------

    p**4

    ✓ Correct
    Feedback:

You know that the probability of getting a red ball in one trial is equal to p. 

Now, for getting 4 red balls in 4 trials, 

you would want to get a red ball in the first trial, 

another one in the second trial, yet another one in the third trial, 

and one final red ball in the fourth trial. 

Hence, the probability would be equal to 

    p*p*p*p = p^4

Generalised Probability
-----------

Now, let’s say you draw a ball from the bag, note its colour, and put it back in the bag. 

Let’s say you perform this activity n times. 

What is the probability that you would get a red ball r times out of n? 

In other words, what will be P(X=r)? 

(Word Limit: 150)

    [Hint: You will need to know the number of combinations 
    in which you could get r red balls and n-r blue balls.]

    [You’re not expected to know the right answer at this point. 
    This question is given just to get you thinking in the right direction.]

----------

Suggested Answer
-------

The number of combinations in which you could get r red balls and n-r blue balls is equal to nCr

So, the final probability will be ​

P ( X = r ) = n C r * ( p ) ** r * ( 1 − p ) ** (n − r)

Binomial Distribution
--------

A student attempts a test with 10 multiple choice questions, where each question has 4 possible options. 

She does not know the correct answer to any of the questions and just randomly selects one of the 4 options for each of them. 

The probability that she would still get 3 questions correct is:

10
C
3
(
1
/
4
)
10

10
C
3
(
3
/
4
)
3
(
1
/
4
)
7

10
C
3
(
1
/
4
)
3
(
3
/
4
)
7


Binomial probability cannot be used here

------------

10
C
3
(
1
/
4
)
3
(
3
/
4
)
7

    ✓ Correct
    Feedback:

There are a total of 4 options. 

Hence, the probability that the student would randomly select the correct option is 1 / 4. 

    Hence, p = 1 / 4

Now, the probability of getting 3 out of 10 questions correct can be found by using the binomial probability for n = 10, r = 3. 

Hence, the probability = 

10
C
3
(
1
/
4
)
3
(
3
/
4
)
7

    = 0.25

In [13]:
com(10, 3) * ((1/4) ** 3) * (1 - 1/4) ** (10 - 3)

0.25028228759765625

In [14]:
n, r, p = 10, 3, 1/4
px(n, r, p)

0.25028228759765625

Binomial Distribution
---------

Now, the test is modified by adding 5 true and false questions to the set of 10 MCQs. 

The student does not know the answer to the new questions either and randomly selects true or false for each of them too.

Now, according to `binomial distribution`, the probability that she would still get 5 of the 15 questions 

(5 true/false Qs, and 10 MCQs with 4 options) correct is:

------------

15
C
5
(
1
/
4
)
5
(
3
/
4
)
10

15
C
5
(
1
/
2
)
5
(
1
/
2
)
10

15
C
5
(
1
/
4
)
3
(
1
/
2
)
2
(
3
/
4
)
10

Binomial probability cannot be used here

---------------

    Binomial probability cannot be used here

    ✓ Correct
    Feedback:

For 10 of the questions (MCQs), the probability of success is p = 1 / 4. 

However, for the remaining 5 questions (T/F questions), the probability of success is p = 1 / 2. 

Hence, you `cannot use binomial probability` here, as the probability of success is not the same in all the trials.

<br>

[![ss](https://upww.screenrec.com/images/f_YzlRXyZbIVhNOen6AtvSJsdfqkQp0129.png)](https://webspace.maths.qmul.ac.uk/i.goldsheid/MTH5118/Notes6-09.pdf)

"Less than" Probability
----------

Recall the UpGrad game with the red and blue balls? 

For this game, can you find the probability of getting 3 or fewer red balls i.e. P(X<3)?

The original probability distribution is given here -

<br>

x (Number of Red Balls) |	P(x)
--- | ---
0	| 0.0256
1	| 0.1536
2	| 0.3456
3	| 0.3456
4	| 0.1296

<br>

    0.8704

    0.1296

    0.7408

-------

    0.8704

    ✓ Correct
    Feedback:

P(X<=3) = the probability of gettingless than or equal to 3 red balls, 

i.e. 0, 1, 2, or 3 red balls. 

So, the required probability 

    = P(X=0) + P(X=1) + P(X=2) + P(X=3)

which evaluates to `0.0256 + 0.1536 + 0.3456 + 0.3456 = 0.8704`

In [15]:
0.0256 + 0.1536 + 0.3456 + 0.3456

0.8704000000000001

Cumulative Probability
---------

Let’s define X as the number of wickets Ishant Sharma would take in the next T20 match he plays. 

Also, the following is an incomplete table for cumulative probability based on previous experience:

<br>

x(Number of wickets taken by Ishant Sharma in a T20 Match)	|  F(x)  
---- | ----
0	| 0.35
1	| 0.55
2	| 0.75
 
<br>

What is the probability that he would take more than 2 wickets in the next  T20 match he plays?

    0.75

    0.25

    0.5

    1

---------

    0.25

    ✓ Correct
    Feedback:

You know that F(2) i.e. P(X<=2) is 0.75

Now, you have to find the probability of X being higher than 2, i.e. P(X>2). 

Notice that the sum of P(X<=2) and P(X>2) would be equal to 1, because their sum will cover all possible outcomes. 

Hence, `P(X<=2) + P(X>2) = 1`, which gives 0.75 + P(X>2) = 1. 

Hence, P(X>2) = 0.25

In [16]:
1 - 0.75

0.25

# Practice Questions

In [17]:
import pandas as pd
df = pd.read_excel('Expected Value for Binomial Distribution.xlsx', 
      index_col = False,
      skiprows=1, 
)

df1 = df.iloc[:, [1, 2, 3]]
df1

r        P(X=r)      r*P(X=r)
0    0  5.987369e-01  0.000000e+00
1    1  3.151247e-01  3.151247e-01
2    2  7.463480e-02  1.492696e-01
3    3  1.047506e-02  3.142518e-02
4    4  9.648081e-04  3.859232e-03
5    5  6.093525e-05  3.046762e-04
6    6  2.672599e-06  1.603559e-05
7    7  8.037891e-08  5.626523e-07
8    8  1.586426e-09  1.269141e-08
9    9  1.855469e-11  1.669922e-10
10  10  9.765625e-14  9.765625e-13

In [18]:
df2 = df.iloc[[0, 2], [5, 6]]
df2.reset_index(drop=True, inplace=True)
df2

n    10
0            p  0.05
1  ∑(r*P(X=r))  0.50

Discrete Probability Distributions
----------

Suppose you take 10 random pasta packets from the market and get them tested for their lead content. 

It just so happens that you have picked out packets from a very defective batch, and there is a `5% (or 0.05)` probability that any pasta packet you select is going to be defective.

----------

What is the probability that 2 of the 10 packets tested would turn out to be defective?

    1.66%

    7.46%

    10%

    15.92%

------------

    7.46%

    ✓ Correct
    Feedback:

Let’s define X as the number of pasta packets found to be defective after testing 10 packets. 

Then, X would follow a binomial distribution with n = 10 and p = 0.05 (Given), the probability of finding a defective packet. 

Now, the probability asked in the question becomes 

P
(
X
=
2
)
=
10
C
2
(
0.05
)
2
(
0.95
)
8


    = 0.0746, or 07.46%

In [19]:
n, r, p = 10, 2, 0.05
x = px(n, r, p)
x

0.07463479852001952

In [20]:
# Ans.

f'{round(x * 100, 2)} %'

'7.46 %'

Discrete Probability Distributions
---------

What is the probability that, after testing these 10 packets, not more than 2 packets would turn out to be defective?

    98.84%

    75.42%

    82.68%

    90.65%

---------

    98.84%

    ✓ Correct
    Feedback:

If X is defined as the number of pasta packets found to be defective after testing 10 packets, then X would follow a binomial distribution with n = 10 and p = 0.05

So, the probability asked in the question becomes 

P(X<=2) = P(X=0) + P(X=1) + P(X=2) 

= 
10
C
0
(
0.95
)
10
+
10
C
1
(
0.05
)
1
(
0.95
)
9
+
10
C
2
(
0.05
)
2
(
0.95
)
8

 = 0.5987 + 0.3151 + 0.0746 
 
    = 0.9884, or 98.84%

In [21]:
n, p = 10, 0.05
s = 0

for l in range(3):
  s += px(n, l, p)

s

0.9884964426207028

In [22]:
# Ans.

f'{round(s * 100, 3)} %'

'98.85 %'

Discrete Probability Distributions
----------

Let’s define X as the number of packets found to be defective after the 10 packets have been tested. 

What will be the expected value of X?

(Hint: You can Use Excel or R for the calculations for this part. 

Use "COMBIN(n,r)" in Excel and "choose(n,r)" in R to find 
n
C
r
)


    1

    0.5

    55

    2

-------

    0.5

    ✓ Correct
    Feedback:

If X is defined as the number of pasta packets found to be defective after testing 10 packets, 

then X would follow a binomial distribution with n = 10 and p = 0.05

Now, you want to find the expected value, which would be equal to 

EV(X) = 0*P(X=0) + 1*P(X=1) + 2*P(X=2) + 3*P(X=3) + 4*P(X=4) + 5*P(X=5) + 6*P(X=6) + 7*P(X=7) + 8*P(X=8) + 9*P(X=9) + 10*P(X=10).

Using a spreadsheet to calculate the individual probabilities, you can find that the expected value EV(X) = 0.5

In [23]:
n, p = 10, 0.05
s = 0

for l in range(n + 1):
  s += l * px(n, l, p)

s

0.4999999999999998

In [24]:
# Ans.

round(s, 1)

0.5

# Comprehension: Expected Value

E
[
X
]
=
∑
(
X
×
P
(
X
)
)

Expected Value
--------

Suppose you’re conducting an experiment by tossing two coins simultaneously. 

You then note down the number of tails in each toss. 

How many tails can you expect on average in each experiment on average?

`Note`: Round off your answer to 2 decimal places. 

If your answer if 1.333333, write 1.33 in the box.

    Ans. 1

--------------

`Solution :`

When two coins are tossed,the sample space is

    S = (H,H),(H,T),(T,H),(T,T)

Let H be the number of tails that occur.

The random variable H take values of 0,1,2

H=0 when no tails occur.or 2 heads occur

    P(H=0)=1/4

H=1 when only one tail occurs, when (H,T) or (T,H) occurs

    P(H=1)=1/2

H=2, when only two tails occur.when (T,T) occurs

    P(H=2)=1/4

Now, we can calculate the expected value of the number of tails:

    E(number of tails) = 0*(1/4) + 1*(1/2) + 2*(1/4)
    = 0 + 1/2 + 1/2 = 1    

Expected Value
--------

Suppose you go to a shoe-shop to buy a pair of shoes. 

There are 50 pairs of shoes in the shop in-total. 

Every time you ask the shopkeeper to show you a pair of shoes, he draws 3 pair of shoes randomly from his stock of 50 pairs. 

If you don’t like any of the pairs, he places all the 3 of them back and then draws 3 pairs again, randomly. 

`Out of all the 50 pairs, 7 pairs of shoes are defected`. (means p = 7/50)

What is the expected number of defective shoes in a given trial?

`Note`: Round off your final answer to 2 decimal places. 

If your answer if 1.333333, write 1.33 in the box.

    Ans. 0.42
    
--------------

`Solution`:

Let X be defective shoes,

    X = {0,1,2,3}

    P(X) = {
      (3C0) * (7/50)**0 * (43/50)**3, 
      (3C1) * (7/50)**1 * (43/50)**2, 
      (3C2) * (7/50)**2 * (43/50)**1, 
      (3C3) * (7/50)**3 * (43/50)**0
    }

Put the above terminal in the formula 

    E[X] = ∑X * P(X) to get 0.42

In [25]:
n, p = 3, 7/50
s = 0

for l in range(n + 1):
  s += l * px(n, l, p)

s

0.42000000000000004

In [26]:
# Ans.

round(s, 2)

0.42

Expected Value
---------

Suppose you’re playing a game using two dice. 

Upon throwing the two dice simultaneously, if their sum equals 4, then you get 1000 rupees. 

If their sum is anything other than 4, you lose 100 rupees. 

What is the expected earning/loss in this game?

`Note`:

1. Round off your final answer to 2 decimal places. 

    If your answer if 1.333333, write 1.33 in the box.

2. If the expectation is a loss, make sure you write a negative number. 

    For instance, -2.5 is a valid answer

---------

`Solution`:

    Sum = 4 when, 
    Dice show : (1, 3), (2, 2), (3, 1)
    p = 3/36

    X = {+1000, -100}
    P(X) = {3/36, 33/36}

    EX = 1000 * 3/36 + (-100) * (1 - 3/36)
       = -8.333333333333329

If you use the expected value formula, you'll get `-8.33` as the answer.

In [27]:
s = 0
x = [+1000, -100]
y = [3/36, 33/36]

for i, j in zip(x, y):
  s += i * j

s

-8.333333333333329

In [28]:
# Ans.

round(s, 2)

-8.33

Expected Value
-----------

Rahul wants to play a poker game. 

The entry charge for the game is a non-refundable INR 2000 and probability that Rahul wins a poker game is 3%. 

The prize money is INR 50000. 

If Rahul wins, he gets the prize money. 

If he loses, he gets nothing. 

What's his expected earning/loss per game?

`Note`:

1. Round off your final answer to 2 decimal places. 

    If your answer if 1.333333, write 1.33 in the box.

2. If the expectation is a loss, make sure you write a negative number. 

    For instance, -2.5 is a valid answer.

--------------



In [29]:
s = 0
e = 2000
x = [50000, 0]
y = [3/100, 1 - 3/100]

for i, j in zip(x, y):
  s += i * j

s

1500.0

In [30]:
# Ans.

ev = s - e
ev

-500.0

Expected Value
-----------

This question is a continuation of the previous question. 

Based on the expected value of the previous question, which of the following is a possibility for Rahul?

`Note`: Multiple answers might be correct.

-------------

Rahul wins INR 50,000 prize money in the first game.

    ✓ Correct
    Feedback:

    If Rahul wins the poker, he'll get INR 50,000. 
    So this is possible.

-------------

Rahul loses all the first 100 games.

    ✓ Correct
    Feedback:

    This is also possible. 
    However, the possibility of this happening is really small.

-------------------

Rahul loses INR 500 rupees in the first game.